include/libcyphal/transport/can/delegate.hpp

```diff
@@ -194,14 +230,19 @@ class TransportDelegate
 
 };  // TransportDelegate
 
+/// This internal session delegate class serves the following purpose: it provides interface
+/// to access a session from transport (by casting canard's `user_reference` member to this class).
+///
 class SessionDelegate
```

Comment:
I would like to keep big-`I`-prefixed stuff for public interfaces. This `SessionDelegate` (in the next PR it will be renamed to `RxSessionDelegate` actually) and similar other `TransportDelegate` are some sort of glue or "gateway" between transport and its sessions - definitely internal stuff, not for use (hence the `detail` namespace).

Reply:
👍